spec/std/file_utils_spec.cr

```diff
@@ -98,5 +98,30 @@ describe "FileUtils" do
         Dir.rmdir(path) if Dir.exists?(path)
       end
     end
+
+    it "doesn't follow symlinks" do
+      data_path = File.join(__DIR__, "data")
+      removed_path = File.join(data_path, "rm_r_test_removed")
+      linked_path = File.join(data_path, "rm_r_test_linked")
+      link_path = File.join(removed_path, "link")
+      file_path = File.join(linked_path, "file")
+
+      begin
+        Dir.mkdir(removed_path)
+        Dir.mkdir(linked_path)
+        File.symlink(linked_path, link_path)
+        File.write(file_path, "")
+
+        FileUtils.rm_r(removed_path)
+        Dir.exists?(removed_path).should be_false
```

Comment:
You don't need to test that it removed a directory (the test above did that), just that the link itself got removed but not the destination folder nor its contents.